src/control/system/raft/raft.go

```diff
@@ -148,10 +148,21 @@ func (db *Database) Barrier() error {
 			db.log.Errorf("lost leadership during Barrier(): %s", err)
 			return errNotSysLeader(svc, db)
 		}
-		return err
+		if err != nil {
+			return err
+		}
+
+		db.waitForLeaderStepUp()
+		return nil
 	})
 }
 
+func (db *Database) waitForLeaderStepUp() {
+	for db.steppingUp.IsTrue() {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
what's the rationale behind the fixed sleep, will this be sufficient for all cases/environments? are there any scenarios which might require a different value? is there any way to introduce a call back rather then using a fixed sleep which might turn out to be brittle? if it really is the only/best option then let's go with it but I wanted to ask the obvious questions first

Reply:
The goal is just to make this less of a tight loop, polling for this bool to change. 10 ms was an arbitrary choice, but the idea was to keep it to a small interval, just not constant checking where it might eat up a lot of cycles.